src/test/java/com/microsoft/sqlserver/jdbc/bulkCopy/BulkCopyRowSetTest.java

```diff
@@ -71,18 +73,44 @@ public void testBulkCopyFloatRowSet() throws SQLException {
         }
     }
 
+    @Test
+    public void testBulkCopyJapaneseCollation() throws SQLException {
+        try (Connection con = getConnection(); Statement stmt = connection.createStatement();
+                SQLServerBulkCopy bulkCopy = new SQLServerBulkCopy(con);) {
+            RowSetFactory rsf = RowSetProvider.newFactory();
+            CachedRowSet crs = rsf.createCachedRowSet();
+            RowSetMetaData rsmd = new RowSetMetaDataImpl();
+            rsmd.setColumnCount(1);
+            rsmd.setColumnName(1, "c1");
+            rsmd.setColumnType(1, java.sql.Types.VARCHAR);
+            rsmd.setTableName(1, tableName2);
+
+            crs.setMetaData(rsmd);
+            crs.moveToInsertRow();
+            crs.updateString("c1", "ああ");
+            crs.insertRow();
+            crs.moveToCurrentRow();
+
+            bulkCopy.setDestinationTableName(tableName2);
+            bulkCopy.writeToServer(crs);

```

Comment:
Need to read from the table here to make sure the data is inserted correctly.

Reply:
Thanks, updated.